File: 974. Subarray Sums Divisible by K/subarraysDivByK.py
class Solution:
    def subarraysDivByK(self, A: List[int], K: int) -> int:
        ans = 0
        sumA = 0
        hashmap = {}
        hashmap[0] = 1
        for n in A:
            sumA += n
            mod = (sumA % K + K) % K
            if not hashmap.__contains__(mod):
                hashmap[mod] = 0
            ans += hashmap[mod]
            hashmap[mod] += 1
        return ans

File: 974. Subarray Sums Divisible by K/subarraysDivByK.java
class Solution {
    public int subarraysDivByK(int[] A, int K) {
        int sum = 0, ans = 0, mod;
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);
        for (int n: A) {
            sum += n;
            mod = (sum % K + K) % K;
            ans += map.getOrDefault(mod, 0);
            map.put(mod, map.getOrDefault(mod, 0) + 1);
        }
        return ans;
    }
}